docs: add info on how to use a specific config

oupala opened the pull request:
Add information about how to install and use this package in different configurations.

Closes #167.

wooorm reviewed (approved):
✨

Arcanemagus commented on README.md:
```diff
@@ -9,31 +9,39 @@
-[remark-preset-lint-recommended][recommended] (both can be turned off).
-You can a also turn on [remark-preset-lint-markdown-style-guide][styleguide].
-By default, [YAML frontmatter][yaml] is supported, but you can turn that off.
+## standard install (default configuration)
```
```suggestion
## Standard Installation
```
Headers should always be capitalized, I'd also leave off the part about the configuration as it's described in detail below.

Arcanemagus commented on README.md:
```diff
@@ -9,31 +9,39 @@
 
-Lint markdown files using [remark-lint][remark-lint] and the
-[linter][linter] package for atom.
+Lint markdown files using [remark-lint][remark-lint] and the [linter][linter] package for atom.
```
```suggestion
Lint markdown files using [remark-lint][remark-lint] and the
[linter][linter] package for Atom.
```
I'd prefer the line splitting to be in there, it makes no difference when rendered but makes diffs _much_ nicer to view.

Also "atom" should be capitalized.

Arcanemagus commented on README.md:
```diff
@@ -9,31 +9,39 @@
-If there *is* configuration for **remark-lint**, through `.remarkrc` files
-or `remarkConfig` in `package.json`s, this linter works just like
-[remark-cli][cli].
+To use this `linter-markdown` package for Atom, simply install the package using the Atom package manager. If there is no configuration found for **remark-lint**, this linter runs [remark-preset-lint-consistent][consistent] and [remark-preset-lint-recommended][recommended] (both can be turned off). You can a also turn on [remark-preset-lint-markdown-style-guide][styleguide]. By default, [YAML fro ...
```
Please just add the sentence at the start, the rest is identical and should be left on individual lines.

oupala replied:
Ok, fixed.

Arcanemagus commented on README.md:
```diff
@@ -9,31 +9,39 @@
-(`npm install` without the `-g` or `--global` flag). If you do install modules
-globally, you must either use [nvm][], or have
-a [`prefix` in your `.npmrc`][prefix].
+If you need a specific configuration for the package, there is 2 ways of installing this `linter-markdown` package: a system-wide way, and a folder-specific way.
```
🔥 How `linter-markdown` is installed doesn't change regardless of how the extensions for `markdown` are installed.

oupala replied:
I changed nothing because I don't understand what to do...

Arcanemagus commented on README.md:
```diff
@@ -9,31 +9,39 @@
+
+You then have to globally install all modules referenced in your `.remarkrc` file, alongside `remark` and `remark-cli` (`npm install` with the `-g` or `--global` flag).
+
+## folder-specific install (project-specific configuration)
```
This should be listed before the global instructions, project specific ones are _always_ preferred as they should be committed to the repository and shared by anyone working on it.

Global installations are a last resort that should virtually never actually be used.

oupala replied:
Ok, fixed.

Arcanemagus commented on README.md:
```diff
@@ -9,31 +9,39 @@
 
-A similar linter, [linter-remark][], runs all remark plugins, but only
-when they are configured to run.
+A similar linter, [linter-remark][], runs all remark plugins, but only when they are configured to run.
```
Again, please don't undo the line breaks.

oupala replied:
Ok, fixed.

Arcanemagus commented on README.md:
```diff
@@ -9,31 +9,39 @@
+
+You then have to globally install all modules referenced in your `.remarkrc` file, alongside `remark` and `remark-cli` (`npm install` with the `-g` or `--global` flag).
+
+## folder-specific install (project-specific configuration)
```
```suggestion
## Project-specific Configuration
```
Let's emphasize that this is related to the configuration, not the method of installation of the plugins. This also has nothing to do with how `linter-markdown` itself is installed so I don't really like the word "install" in the header here as it could confuse people.

Arcanemagus commented on README.md:
```diff
@@ -9,31 +9,39 @@
+
+You can set you folder-specific configuration in a `.remarkrc` file that should be saved in each folder of your markdown documents.
+
+You then have to locally install all modules referenced in your `.remarkrc` file, alongside `remark` and `remark-cli` (`npm install` without the `-g` or `--global` flag). If you do install modules globally, you must either use [nvm][], or have a [`prefix` in your `.npmrc`][prefix].
```
```suggestion
You then have to locally install all modules referenced in your `.remarkrc`
file, alongside `remark` and `remark-cli` (`npm install` without the `-g` or
`--global` flag).
```
The removed part is only relevant for a global installation.

Arcanemagus commented on README.md:
```diff
@@ -9,31 +9,39 @@
+
+If there *is* configuration for **remark-lint**, through `.remarkrc` files or `remarkConfig` in `package.json`s, this linter works just like [remark-cli][cli].
+
+See [this tutorial][set-up] on how to set-up npm to work without sudo, which is good practise, and comes with the added benefit that `linter-markdown` can pick up on globally installed modules.
```
```suggestion
See [this tutorial][set-up] on how to set-up `npm` to work without `sudo`, which
is good practice, and comes with the added benefit that `linter-markdown` can
pick up on globally installed modules.
```
Misspelling of "practise" -> "practice" and no wrapping.

Arcanemagus commented on README.md:
```diff
@@ -9,31 +9,39 @@
+
+You then have to locally install all modules referenced in your `.remarkrc` file, alongside `remark` and `remark-cli` (`npm install` without the `-g` or `--global` flag). If you do install modules globally, you must either use [nvm][], or have a [`prefix` in your `.npmrc`][prefix].
+
+## other informations
```
```suggestion
## More Information
```

Arcanemagus commented on README.md:
```diff
@@ -9,31 +9,39 @@
-See [this tutorial][set-up] on how to set-up npm to work without sudo, which
-is good practise, and comes with the added benefit that `linter-markdown`
-can pick up on globally installed modules.
+## system-wide install (system-wide configuration)
```
```suggestion
## Global Configuration
```
This entire section should be de-emphasized below the project specific configuration.

Arcanemagus commented on README.md:
```diff
@@ -9,31 +9,39 @@
-can pick up on globally installed modules.
+## system-wide install (system-wide configuration)
+
+The system-wide way of installing the `linter-markdown` package should be used if you use the same configuration on all your markdown documents. This solution is also convenient if your markdown documents are not part of an npm project (and hence do not have an associated package.json file).
```
```suggestion
The global configuration should only be used if you need to use the same
configuration on all your Markdown documents, and they aren't contained within
projects of their own.
```
Global configurations are virtually never the right solution, they should only be used if:
* The files aren't in projects **and**
* The files will _never_ be worked on by anyone else (even if the "other person" is yourself later on in the future)

And as stated before this has nothing to do with how `linter-markdown` itself is installed.

Arcanemagus commented on README.md:
```diff
@@ -9,31 +9,39 @@
+
+The system-wide way of installing the `linter-markdown` package should be used if you use the same configuration on all your markdown documents. This solution is also convenient if your markdown documents are not part of an npm project (and hence do not have an associated package.json file).
+
+You can set you system-wide configuration in a `.remarkrc` file that should be saved in `~/.remarkrc`, at the root of your home directory.
```
```suggestion
You can set you system-wide configuration in a `.remarkrc` file that should be
saved in `~/.remarkrc`, at the root of your home directory.
```

Arcanemagus commented on README.md:
```diff
@@ -9,31 +9,39 @@
+
+You can set you system-wide configuration in a `.remarkrc` file that should be saved in `~/.remarkrc`, at the root of your home directory.
+
+You then have to globally install all modules referenced in your `.remarkrc` file, alongside `remark` and `remark-cli` (`npm install` with the `-g` or `--global` flag).
```
```suggestion
You then have to globally install all modules referenced in your `.remarkrc`
file, alongside `remark` and `remark-cli` (`npm install` with the `-g` or
`--global` flag).
```

The line break between this sentence and the last should probably also be removed.

Arcanemagus commented on README.md:
```diff
@@ -9,31 +9,39 @@
+
+## folder-specific install (project-specific configuration)
+
+The folder-specific way of installing the `linter-markdown` package should be used if you use differents configurations depending of each markdown document. This solution is also convenient if your markdown documents are part of an npm project (and hence have an associated package.json file).
```
```suggestion
The folder-specific way of configuring `remark` should be used if:
* You are working on the project with other people
* You need repeatable results
* You use need a different configuration for each project
```

Arcanemagus commented on README.md:
```diff
@@ -9,31 +9,39 @@
+
+The folder-specific way of installing the `linter-markdown` package should be used if you use differents configurations depending of each markdown document. This solution is also convenient if your markdown documents are part of an npm project (and hence have an associated package.json file).
+
+You can set you folder-specific configuration in a `.remarkrc` file that should be saved in each folder of your markdown documents.
```
```suggestion
You can set you folder-specific configuration in:
* A `remarkConfig` section in `package.json`
* A `.remarkrc.js` or `.remarkrc` file
These should be placed in your project, see
[remark configuration](https://github.com/unifiedjs/unified-engine/blob/master/doc/configure.md)
for details.
```
The link there can probably be put down with the other links.

Arcanemagus commented on README.md:
```diff
@@ -9,31 +9,39 @@
+
+## other informations
+
+If there *is* configuration for **remark-lint**, through `.remarkrc` files or `remarkConfig` in `package.json`s, this linter works just like [remark-cli][cli].
```
```suggestion
If there *is* configuration for **remark-lint**, through `.remarkrc` files or
`remarkConfig` in `package.json`s, this linter works just like
[remark-cli][cli].
```

Arcanemagus reviewed (changes requested):
Here's a first pass review of this.

If you have trouble wrapping the lines Atom has a built in command to reflow selected text for you:
* `Autoflow: Reflow Selection` in the Command Palette
* <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>Q</kbd> on Windows/Linux

Arcanemagus committed:
refactor: capitalize header

Co-Authored-By: oupala <oupala@users.noreply.github.com>

Arcanemagus committed:
refactor: line splitting and Atom capitalization

Co-Authored-By: oupala <oupala@users.noreply.github.com>

Arcanemagus committed:
refactor: line splitting

Co-Authored-By: oupala <oupala@users.noreply.github.com>

Arcanemagus committed:
refactor: line splitting

Co-Authored-By: oupala <oupala@users.noreply.github.com>

Arcanemagus committed:
refactor: update README.md

Co-Authored-By: oupala <oupala@users.noreply.github.com>

Arcanemagus committed:
refactor: line splitting

Co-Authored-By: oupala <oupala@users.noreply.github.com>

Arcanemagus committed:
refactor: update README.md

Co-Authored-By: oupala <oupala@users.noreply.github.com>

Arcanemagus committed:
refactor: update README.md

Co-Authored-By: oupala <oupala@users.noreply.github.com>

Arcanemagus committed:
refactor: typoe and line splitting

Co-Authored-By: oupala <oupala@users.noreply.github.com>

Arcanemagus committed:
refactor: capitalize header

Co-Authored-By: oupala <oupala@users.noreply.github.com>

Arcanemagus committed:
refactor: update README.md

Co-Authored-By: oupala <oupala@users.noreply.github.com>

Arcanemagus committed:
refactor: update README.md

Co-Authored-By: oupala <oupala@users.noreply.github.com>

Arcanemagus committed:
refactor: update README.md

Co-Authored-By: oupala <oupala@users.noreply.github.com>

oupala committed:
refactor: resolving conversations

oupala committed:
refactor: moving project-specific configuration before global one

Moving project-specific configuration before global configuration.

renovate-bot committed:
chore(deps): update dependency eslint to v5.15.0

renovate-bot committed:
chore(deps): update dependency eslint to v5.15.1

oupala commented:
@Arcanemagus I've put a lot of effort in this pull request (created from a previous one). I also changed pretty all what you asked to change. Could you please tell me if this pull request is mergeable or not?

renovate-bot committed:
chore(deps): update dependency eslint to v5.15.3

oupala commented:
@Arcanemagus I had to update the pull request one more time. Could you please merge this pull request or tell me what should be improved?

renovate-bot committed:
chore(deps): update dependency eslint to v5.16.0

oupala commented:
I'm a bit desperate about this pull request.

Should it be **that** hard to contribute to a *bit* of documentation?

renovate-bot committed:
chore(deps): update dependency eslint-plugin-import to v2.17.1

renovate-bot committed:
chore(deps): update dependency eslint-plugin-import to v2.17.2

renovate-bot committed:
fix(deps): update dependency atom-package-deps to v5.1.0

semantic-release-bot committed:
chore(release): 5.2.7 [skip ci]

renovate-bot committed:
chore(deps): update dependency husky to v2

renovate-bot committed:
chore(deps): update dependency husky to v2.1.0

renovate-bot committed:
chore(deps): update dependency husky to v2.2.0

renovate-bot committed:
chore(deps): update commitlint monorepo to v7.6.0

renovate-bot committed:
chore(deps): update dependency semantic-release to v15.13.12

renovate-bot committed:
chore(deps): update commitlint monorepo to v7.6.1

renovate-bot committed:
chore(deps): update dependency husky to v2.3.0

renovate-bot committed:
chore(deps): update dependency eslint-plugin-import to v2.17.3

renovate-bot committed:
chore(deps): update commitlint monorepo to v8

renovate-bot committed:
chore(deps): update dependency husky to v2.4.0

renovate-bot committed:
chore(deps): update dependency husky to v2.4.1

renovate-bot committed:
chore(deps): update dependency semantic-release to v15.13.16

renovate-bot committed:
fix(deps): update dependency remark-frontmatter to v1.3.2

semantic-release-bot committed:
chore(release): 5.2.8 [skip ci]

renovate-bot committed:
fix(deps): update remark packages

semantic-release-bot committed:
chore(release): 5.2.9 [skip ci]

renovate-bot committed:
chore(deps): update dependency husky to v2.5.0

renovate-bot committed:
chore(deps): update dependency eslint to v6

renovate-bot committed:
chore(deps): update eslint packages

renovate-bot committed:
chore(deps): update dependency semantic-release to v15.13.17

renovate-bot committed:
chore(deps): update dependency husky to v2.6.0

renovate-bot committed:
chore(deps): update dependency semantic-release to v15.13.18

renovate-bot committed:
chore(deps): update dependency husky to v2.7.0

renovate-bot committed:
chore(deps): update dependency eslint-config-airbnb-base to v13.2.0

renovate-bot committed:
fix(deps): update remark packages

renovate-bot committed:
chore(deps): update dependency eslint-config-airbnb-base to v14

renovate-bot committed:
chore(deps): update dependency semantic-release to v15.13.24

renovate-bot committed:
chore(deps): update eslint packages

renovate-bot committed:
chore(deps): update dependency husky to v3

Arcanemagus committed:
chore(deps): update commitlint monorepo to v8.2.0

Arcanemagus committed:
style: fix linting issues

Arcanemagus committed:
fix(deps): update dependency unified-engine-atom to v8

Arcanemagus committed:
ci(appveyor): update Node.js to v10

semantic-release-bot committed:
chore(release): 5.2.10 [skip ci]

renovate-bot committed:
chore(deps): update semantic-release monorepo

renovate-bot committed:
chore(deps): update dependency husky to v3.1.0

oupala commented:
It seems so hard to contribute.

Have I missed something?

renovate-bot committed:
chore(deps): update eslint packages

renovate-bot committed:
fix(deps): update remark packages

semantic-release-bot committed:
chore(release): 5.2.11 [skip ci]

renovate-bot committed:
chore(deps): update dependency semantic-release to v15.14.0

renovate-bot committed:
chore(deps): update dependency eslint to v6.8.0

renovate-bot committed:
chore(deps): update commitlint monorepo to v8.3.3

renovate-bot committed:
chore(deps): update commitlint monorepo to v8.3.4

renovate-bot committed:
chore(deps): update dependency husky to v4

renovate-bot committed:
chore(deps): update dependency husky to v4.0.1

renovate-bot committed:
chore(deps): update dependency husky to v4.0.3

renovate-bot committed:
chore(deps): update dependency husky to v4.0.6

renovate-bot committed:
chore(deps): update dependency semantic-release to v16

renovate-bot committed:
chore(deps): update dependency husky to v4.0.9

renovate-bot committed:
chore(deps): update semantic-release monorepo

renovate-bot committed:
chore(deps): update dependency semantic-release to v17.0.2

renovate-bot committed:
chore(deps): update commitlint monorepo to v8.3.5

renovate-bot committed:
chore(deps): update dependency eslint-plugin-import to v2.20.1

renovate-bot committed:
chore(deps): update dependency husky to v4.2.1

renovate-bot committed:
chore(deps): update dependency husky to v4.2.2

renovate-bot committed:
chore(deps): update dependency semantic-release to v17.0.3

renovate-bot committed:
chore(deps): update dependency husky to v4.2.3

oupala commented:
Review requested for @Arcanemagus.

renovate-bot committed:
chore(deps): update dependency semantic-release to v17.0.4

oupala commented:
@Arcanemagus Would you mind reviewing this merge request? Thanks.

oupala committed:
docs: add info on how to use a specific config

Add information about how to install and use this package in different configurations.

oupala committed:
Merge branch 'master' into master
